### PR TITLE
[backend] fix(hubspot): allow reach out sales for all trial statuses (#1615)

### PR DESCRIPTION
**Pull request by jbanety**
# Context
The "Reach out to Sales" button in trials was only working when the trial status was Active or Expired. This was due to backend filtering that prevented loading trial deployment requests for other statuses. 

We decided that users should be able to contact sales at any point during their trial lifecycle (queued, pending, provisioning, active, expired, failed, cancelled), so we removed the status restriction.

## How to test
1. Create a trial with any status (e.g., Pending, Provisioning, Queued)
2. Try to use the "Reach out to Sales" functionality
3. Verify the HubSpot webhook is called successfully with the correct payload
4. Test from OpenCTI trial app using platform token authentication
5. Test from portal with user authentication

## Related issues

- Related to #1615

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [x] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

### CODE REVIEW

File: apps/portal-api/src/thirdparty/hubspot/hubspot.ts
Comment (carinelebas) [inline apps/portal-api/src/thirdparty/hubspot/hubspot.ts]: We should check here that deployment request is defined. If we don't find a matching deployment request in DB, it will not be, and we'll have an exception. It would be cleaner to properly log this case. Wdyt?

File: apps/portal-api/src/thirdparty/hubspot/hubspot.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export const hubspotReachOutSalesHook = async (
     let deploymentRequest: FullyQualifiedDeploymentRequest;
     if (user?.id) {
       deploymentRequest =
-        await DeploymentRequestDomain.loadProvisionedTrialDeploymentRequestByPlatformIdentifier(
+        await DeploymentRequestDomain.loadTrialDeploymentRequestByPlatformIdentifier(
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/thirdparty/hubspot/hubspot.ts]: It's true the order by should cover this edge case. If you ask for my opinion, indeed I like the clean solution better, it could avoid potential other side effects 😄. Maybe it could be done in a separate issue? (it would be an easy first issue 😉  )
